framework/db/ActiveQuery.php

```diff
@@ -126,7 +128,9 @@ public function init()
      * Executes query and returns all results as an array.
      * @param Connection|null $db the DB connection used to create the DB command.
      * If null, the DB connection returned by [[modelClass]] will be used.
-     * @return array|ActiveRecord[] the query results. If the query results in nothing, an empty array will be returned.
+     * @return T[] the query results. If the query results in nothing, an empty array will be returned.
```

Comment:
That will likely break many IDEs. It will likely break API docs generation as well. Can we keep regular type for `@return` and add psalm/phpstan annotations only?

Reply:
I think it's possible